modules/ROOT/pages/dataweave-language-guide.adoc

```diff
@@ -1,75 +1,212 @@
+DataWeave enables you to take advantage of the benefits of functional programming, including:
 
-//LINK TO DW 1.0 LANDING PAGE IN MULE 3.9 DOCS:
-include::partial$dataweave1-links.adoc[tag=dataweave1LandingPage]
+* Pure functions that always produce the same output given the same input, making them easier for you to debug.
+* Immutable variables that do not change their value during the execution, adding stability to your code.
+* Function signatures that provide additional transparency about what affects a function's execution, because pure functions depend on only their input parameters to produce a result.
+* Functional languages produce code that’s easier to maintain because the code is concise and clear.
+* A call-by-need strategy (also known as _lazy evaluation_) that improves performance, because expressions are evaluated only when they are needed and their result is stored to avoid evaluating an expression multiple times if the input parameters are the same.
 
-.Input Payload
-[source,json,linenums]
+== DataWeave Language Basic Concepts
+
+DataWeave incorporates concepts that are common to most programming languages. All DataWeave scripts consist of a _header_ that contains important directives, including the output format of the transformation, and a _body_ that contains the expression to generate a result.
+
+DataWeave supports different _data structures_, including _simple_, _complex_, and _composite_ types. In most of your DataWeave transformations, you use _data selectors_ and _functions_ alongside _arrays_ and _objects_ to perform data mappings or transformations.
+
+You can also import prebuilt DataWeave _modules_ to extend the set of functions available to use in your scripts.
+
+=== Architecture of a DataWeave Script
+
+A DataWeave script consists of a _header_ and a _body_, separated by three dashes (`---`).
+The header contains language directives (`import`, for example), defines the output format of the transformation, and can also contain variable and function declarations.
+The body contains the expression that produces the resulting output, usually a data mapping or a transformation.
+
+Here is an example of a basic DataWeave transformation:
+[source, dataweave, linenums]
+----
+%dw 2.0
+output application/json
+---
+payload
+----
+
+=== Data Types
+
+DataWeave represents data using values, each of which has a data type associated with it.
+The type system enables you to apply constraints to variables and function parameters. The supported data types include simple types such as `String`, `Boolean`, `Number`, and `Date`, as well as composite and complex types such as `Array`, `Object`, and `Any`.
+
+For a complete list of all supported data types, see xref:dataweave-type-system.adoc[Type System].
+
+=== Data Selectors
+
+Data selectors enable you to access fields within a data structure so that you can retrieve data from the payload or any other variable created during the DataWeave script execution.
```

Comment:
Do we reference info on what the payload is? It's mentioned a few times in this article.